test/new-e2e/tests/cws/ec2_test.go

```diff
@@ -241,6 +244,36 @@ func (a *agentSuite) Test03OpenSignal() {
 	}, 4*time.Minute, 10*time.Second)
 }
 
+func (a *agentSuite) Test04SecurityAgentSIGTERM() {
+	output := a.Env().RemoteHost.MustExecute("cat /opt/datadog-agent/run/security-agent.pid")
+	pid, err := strconv.ParseInt(strings.TrimSpace(output), 10, 64)
+	require.NoError(a.T(), err, "failed to parse security-agent pid")
+
+	var start, end time.Time
+	start = time.Now()
+	_, err = a.Env().RemoteHost.Execute(fmt.Sprintf("sudo kill -SIGTERM %d", pid))
+	require.NoError(a.T(), err, "failed to send SIGTERM to security-agent")
+	exited := assert.EventuallyWithT(a.T(), func(c *assert.CollectT) {
+		_, err := a.Env().RemoteHost.Execute("pgrep -x security-agent")
+		if err == nil {
+			c.Errorf("security-agent should not be running")
+			return
+		}
+
+		// pgrep exits with 1 if no process is found
+		var exitErr *ssh.ExitError
+		if errors.As(err, &exitErr) && exitErr.ExitStatus() == 1 {
+			end = time.Now()
+			return
+		}
+		assert.NoError(c, err, "failed to check the security-agent process state")
+	}, 30*time.Second, 1*time.Second)
+
+	if exited {
+		a.T().Logf("security-agent exited after %s", end.Sub(start).String())
+	}
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Should you restart the security agent after this test? Could it impact other subtests in the suite?

Reply:
The order of the sub-tests currently makes it so that the security-agent doesn't need to run after this newly added sub-test. But I agree this is a bad pattern so I'll force a restart of the security-agent at the end of this test, thanks!